sklearn/tests/test_base.py

```diff
@@ -1081,3 +1081,56 @@ def test_param_is_default(default_value, test_value):
     estimator = make_estimator_with_param(default_value)(param=test_value)
     non_default = estimator._get_params_html().non_default
     assert "param" not in non_default
+
+
+def test_is_regressor_class_vs_instance_error():
+    """Test is_regressor gives helpful error when passed a class instead of instance."""
+    # Test using a standard library class that has __init__ but no __sklearn_tags__
+    with pytest.raises(TypeError, match="Expected an estimator instance"):
+        is_regressor(dict)
```

Comment:
This test is not really what we want to test because `dict` doesn't have `__sklearn_tags__` at all. What we want is distinguish between `Ridge()` and `Ridge` for instance.

Also you can parametrize this test to be a bit less redundant, like
```py
@pytest.mark.parametrize(
    "is_func, Estimator",
    [(is_regressor, Ridge), (is_classifier, LogisticRegression), (is_clusterer, KMeans)],
)
```